src/components/cart.js

```diff
@@ -78,10 +79,28 @@ export default class Cart extends Component {
   get lineItemsHtml() {
     return this.lineItemCache.reduce((acc, lineItem) => {
       const data = Object.assign({}, lineItem, this.options.viewData);
+      const fullPrice = data.variant.priceV2.amount * data.quantity;
+      const formattedPrice = formatMoney(fullPrice, this.moneyFormat);
+      const discountAllocations = data.discountAllocations;
+
+      let totalDiscount = 0;
```

Comment:
Since you're using a reduce below - you could use the resulting "object" to store your total. This way the external `totalDiscount` won't update via a reduce function.